sqlmesh/core/snapshot/definition.py

```diff
@@ -941,6 +941,7 @@ def _model_metadata_hash(model: Model, audits: t.Dict[str, Audit]) -> str:
         str(model.batch_size) if model.batch_size is not None else None,
         json.dumps(model.mapping_schema, sort_keys=True),
         *model.tags,
+        *model.primary_key
```

Comment:
I wonder whether should be a part of the data hash instead. For instance we can use this to set the actual primary key constraint for databases that support this (eg. postgres), in which case we'd like want a new table version.

Reply:
@tobymao wdyt?

Reply:
it is explicitly not an actual primary key, which is why i want it to be called grain and only used for metadata